\documentclass[ignorenonframetext,10pt]{beamer}

%\usepackage[latin1]{inputenc}
\usepackage[english]{babel}
\usepackage{amsmath,amssymb}
\usepackage{multimedia}
\usepackage{alltt}
\usepackage{multirow}
\usepackage{textcomp}
\usepackage[footnotesize]{subfigure}
\usepackage{graphicx}

\usetheme{Darmstadt}
\useoutertheme[subsection=false]{miniframes}
\useoutertheme{smoothbars}
\usefonttheme{structurebold}
\setbeamertemplate{navigation symbols}{}
\setbeamercovered{invisHelices specialized RNAShapeslized RNAShapes}
\author{\large Bj\"orn~Vo\ss}
\date{4th May, 2006}

\institute[ExpBI]{\normalsize
  AG Experimentelle Bioinformatik (Cyanolab)\\
  Institut f\"ur Biologie II\\
  Universit\"at Freiburg\\
  \bigskip
  Oberseminar Bioinformatik\\
  Lehrstuhl f\"ur Bioinformatik\\
  Universit\"at Freiburg}

\begin{document}

\frame{\maketitle}

\section{Introduction}
\subsection{}
\frame
{
  \frametitle{\large Structure Prediction}
  \small
  \begin{block}{Single Sequence (Free energy algorithms)}
    \begin{itemize}
      \item \alert<2->{Minimum free energy structure}
      \item \alert<3->{Suboptimal folding}
      \item \alert<3->{Stochastic sampling}
      \item \alert<3->{Shape abstraction}
      \item \alert<3->{Shape probabilities}
      \item \ldots
    \end{itemize}
  \end{block}
  \begin{block}{Comparative}
    \begin{itemize}
      \item Alignment of folded RNAs (MARNA, RNAforester, \ldots)
      \item Simultaneous aligning and folding (Foldalign, Dynalign, \ldots)
      \item \alert<2->{Folding aligned RNAs (RNAalifold)}
    \end{itemize}
  \end{block}
  \begin{block}{RNAlishapes}<3->
    \alert<3->{Structural Analysis of Aligned RNAs}
  \end{block}
}

\frame{
  \frametitle{\large Starting point}
  \small
  \begin{block}{RNAshapes Version 2.0 (Vo\ss~B., Giegerich R. and Rehmsmeier M., 2006)}
    \begin{itemize}
      \item \textbf{Unambiguous grammar with unique dangles}
      \item Shape abstraction (Version 1.0)
      \item Probabilistic shape analysis
      \item Boltzmann-weighted sampling
      \item Suboptimal folding with correct energies
    \end{itemize}
  \end{block}
  \begin{block}{RNAalifold (Hofacker I.L., Fekete M., and Stadler P.F., 2002)}
    \begin{itemize}
      \item Structure prediction for aligned RNAs
      \item Scoring based on free energy and covariance contribution
    \end{itemize}
  \end{block}
  \begin{block}{Methodology}
    \begin{itemize}
      \item Implemented in ADP 
      \item Alignments as input -- What to do with gaps?
      \item Score as in RNAalifold
    \end{itemize}
  \end{block}
}

\section{Shape Abstraction and ADP}
\subsection{}

\frame
{
  \frametitle{\large Shape Abstraction}
  \small
  \begin{columns}
    \begin{column}{.75\textwidth}
      \begin{block}{Example: Part of Structure Space of tRNA}
        \begin{alltt}\tiny
          GGGCCCAUAGCUCAGUGGUAGAGUGCCUCCUUUGCAAGGAGGAUGCCCUGGGUUCGAAUCCCAGUGGGUCCA\\
          (((((((((((((((.((((.....(((((((...))))))).))))))))))).........)))))))).\\
          (((((((((((((((.((((.....((((((.....)))))).))))))))))).........)))))))).\\
          (((((((((((((((.(((......(((((((...)))))))..)))))))))).........)))))))).\\
          (((((((((((((((.(((......((((((.....))))))..)))))))))).........)))))))).\\
          (((((((.(((((((.((((.....(((((((...))))))).)))))))))))..........))))))).\\
          (((((((.(((((((.((((.....((((((.....)))))).)))))))))))..........))))))).\\
          (((((((((((((((.(((...((.((((((.....)))))))))))))))))).........)))))))).\\
          (((((((..((((((.((((.....(((((((...))))))).))))))))))...........))))))).\\
          (((((((..((((((.((((.....((((((.....)))))).))))))))))...........))))))).\\
          (((((((((((((((.(((......(((((.......)))))..)))))))))).........)))))))).\\
          (((((((.(((((((.((((.....(((((.......))))).)))))))))))..........))))))).\\
          (((((((((((((((.(((...((.(((((.......))))))))))))))))).........)))))))).\\
          \alert<2->{\textbf<2->{((((((((.....((.((((.....(((((((...))))))).))))))(((.......))).)))))))).}}\\
          \alert<2->{\textbf<2->{((((((((.....((.((((.....((((((.....)))))).))))))(((.......))).)))))))).}}\\
          ((((((((..(((((.((((.....(((((((...))))))).)))))))))...........)))))))).\\
          ((((((((..(((((.((((.....((((((.....)))))).)))))))))...........)))))))).\\
          (((((((((((((((.((....((.(((((((...))))))))).))))))))).........)))))))).\\
          (((((((((((((((.((....((.((((((.....)))))))).))))))))).........)))))))).\\
          (((((((((((.(((.((((.....(((((((...))))))).))))))).))).........)))))))).\\
          (((((((((((.(((.((((.....((((((.....)))))).))))))).))).........)))))))).\\
          \alert<2->{\textbf<2->{((((((..........((((.....(((((((...))))))).))))(((((.......))))).)))))).}}\\
          \alert<2->{\textbf<2->{((((((..........((((.....((((((.....)))))).))))(((((.......))))).)))))).}}\\
          \alert<2->{\textbf<2->{(((((((.........((((.....(((((((...))))))).))))(((((.......)))))))))))).}}\\
          \alert<2->{\textbf<2->{(((((((.........((((.....((((((.....)))))).))))(((((.......)))))))))))).}}\\
          ((((((..(((((((.((((.....(((((((...))))))).)))))))))))...........)))))).\\
          ((((((..(((((((.((((.....((((((.....)))))).)))))))))))...........)))))).\\
          ((((((((.((((((.(((......(((((((...)))))))..)))))))))..........)))))))).\\
          \textcolor<2->{blue}{\textbf<2->{((((((...((((.......)))).(((((((...))))))).....(((((.......))))).)))))).}}\\
          \textcolor<2->{blue}{\textbf<2->{((((((...((((.......)))).((((((.....)))))).....(((((.......))))).)))))).}}\\
          (((((((.(((((((.(((......(((((((...)))))))..))))))))))..........))))))).\\
          (((((((.(((((((.(((......((((((.....))))))..))))))))))..........))))))).\\
        \end{alltt}
      \end{block}
    \end{column}
    \begin{column}{.25\textwidth}
      \begin{block}<2->{Shapes}
        \centering
        \vspace{0.2cm}
        \includegraphics<2->[scale=1]{images/tRNA_shape1}
        \vspace{0.2cm}
        \includegraphics<2->[scale=1]{images/tRNA_shape2}
        \vspace{0.2cm}
        \includegraphics<2->[scale=1]{images/tRNA_shape3}
      \end{block}
    \end{column}
  \end{columns}
}

\frame
{
        \frametitle{\large Abstract Shapes of RNA}
        \small
        \begin{block}{Classes of similar structures represented by:}
                \begin{itemize}
                        \item Shape notation:\\
                                                Abstract from helix length and length of unpaired regions\\
                                                '[' and ']': paired regions, '\_': unpaired regions                                             
                        \item \emph{Shrep} (\textbf{sh}ape \textbf{rep}resentative structure):\\
                                                Shape member with lowest free energy
                \end{itemize}
        \end{block}
        \begin{block}{Shape notation - Different abstraction levels}
                \begin{tabular}{cl}
                &\texttt{\scriptsize\textcolor{red}{(((..}\textcolor{green}{((((.....))))}\textcolor{red}{.}\textcolor{blue}{((((...))))}\textcolor{red}{..)))}...\textcolor[gray]{.5}{((..}\textcolor{orange}{(((.....)))}\textcolor[gray]{.5}{..))}..}\\
                Level 1&\texttt{\textcolor{red}{[\_}\textcolor{green}{[\_]}\textcolor{red}{\_}\textcolor{blue}{[\_]}\textcolor{red}{\_]}\_\textcolor[gray]{.5}{[\_}\textcolor{orange}{[\_]}\textcolor[gray]{.5}{\_]}\_}\\
                Level 2&\texttt{\textcolor{red}{[}\textcolor{green}{[]}\textcolor{blue}{[]}\textcolor{red}{]}\_\textcolor[gray]{.5}{[}\textcolor{orange}{[]}\textcolor[gray]{.5}{]}\_}\\
                Level 
3&\texttt{\textcolor{red}{[}\textcolor{green}{[]}\textcolor{blue}{[]}\textcolor{red}{]}\textcolor[gray]{.5}{[}\textcolor{orange}{[]}\textcolor[gray]{.5}{]}}\\
                Level 4&\texttt{\textcolor{red}{[}\textcolor{green}{[]}\textcolor{blue}{[]}\textcolor{red}{]}\_\textcolor{orange}{[]}\_}\\
                \structure<2->{Level 5} & \structure<2->{ \texttt{\textcolor{red}{[}\textcolor{green}{[]}\textcolor{blue}{[]}\textcolor{red}{]}\textcolor{orange}{[]}}}\\
                \end{tabular}
        \end{block}

        \begin{alertblock}<2->{Wanted}
                Predict shapes together with their shreps without calculating all structures
        \end{alertblock}
}


\begin{frame}[fragile]
       \frametitle{\large ADP grammar for RNA folding}
       \small
       \begin{block}{Derive candidates for evaluation (Wuchty, no dangling bases)}
               \scriptsize
               \begin{semiverbatim}
\textcolor{red}{struct}  =  \textcolor{blue}{str  <<<} \textcolor{red}{comps        |||}
           \textcolor{blue}{str  <<<} \textcolor{red}{singlestrand |||}
           \textcolor{blue}{nil  <<<} \textcolor{red}{empty}        \textcolor{violet}{... h}    \uncover<2->{\textleftarrow apply choice function}

\textcolor{red}{block}   = \textcolor{orange}{tabulated}(                      \uncover<3->{\textleftarrow store results in table}
                    \textcolor{red}{closed            |||}
           \textcolor{blue}{blk  <<<} \textcolor{red}{region ~~~ closed} \textcolor{violet}{... h})

\textcolor{red}{comps}   = \textcolor{orange}{tabulated}(
           \textcolor{blue}{cons <<<} \textcolor{red}{block  ~~~ comps        |||}
           \textcolor{blue}{ul   <<<} \textcolor{red}{block                   |||}
           \textcolor{blue}{cons <<<} \textcolor{red}{block  ~~~ singlestrand} \textcolor{violet}{... h})

\textcolor{red}{closed}  = \textcolor{orange}{tabulated}(
          (\textcolor{blue}{hl <<<} \textcolor{red}{base ~~~            region             ~~~ base  |||}
           \textcolor{blue}{sr <<<} \textcolor{red}{base ~~~            closed             ~~~ base  |||}
           \textcolor{blue}{bl <<<} \textcolor{red}{base ~~~      region ~~~ closed        ~~~ base  |||}
           \textcolor{blue}{br <<<} \textcolor{red}{base ~~~      closed ~~~ region        ~~~ base  |||}
           \textcolor{blue}{ml <<<} \textcolor{red}{base ~~~      block  ~~~ comps         ~~~ base  |||}
           \textcolor{blue}{il <<<} \textcolor{red}{base ~~~ region ~~~ closed  ~~~ region ~~~ base}    )
           \textcolor{orange}{`with` basepairing} \uncover<4->{\textleftarrow check for basepairing}              \textcolor{violet}{... h})
                               
\textcolor{red}{singlestrand} =  \textcolor{blue}{ss   <<<} \textcolor{red}{region} 
               \end{semiverbatim}
       \end{block}
\end{frame}

\frame
{
  \frametitle{\large Scoring with Algebras}
  \small
  \begin{block}{Example candidate}
    $\textcolor{red}{SR(1 (}\textcolor{blue}{SR (2 (}\textcolor{green}{SR (3 (}\textcolor{orange}{HL (4 (5,6,7,8) 9)}\textcolor{green}{) 10)}\textcolor{blue}{ 11)}\textcolor{red}{ 12)}\ \Rightarrow$ \texttt{\textcolor{red}{(}\textcolor{blue}{(}\textcolor{green}{(}\textcolor{orange}{(....)}\textcolor{green}{)}\textcolor{blue}{)}\textcolor{red}{)}}\\
    {\footnotesize Note: The candidate is composed of operators and indexes and contains no sequence information}
  \end{block}
  \scriptsize
  \begin{columns}
    \begin{column}{.20\textwidth}
      \begin{block}{\small Function}\tabcolsep1mm
        \begin{tabular}{l@{\hspace{1mm}}l@{\hspace{1mm}}l}
          \textit{HL} &a r b&=\\
          \textit{SR} &a x b&=\\
          \textit{BL} &a r x b&=\\
          \textit{BR} &a x r b&=\\
          \textit{IL} &a r x r` b&=\\
          \textit{ML} &a x b&=\\
          \textit{AD} &x x'&=\\
          \textit{SS} &r&=\\
          \hline
          \phantom{x}&\phantom{x}&\phantom{x}\\
          \textit{h}&\phantom{x}&=
        \end{tabular}
      \end{block}
    \end{column}
    \begin{column}{.33\textwidth}
      \begin{block}{\small Energy}\tabcolsep1mm
        \begin{tabular}{l}
          stackE(a,b) + unpE(r)\\
          x + stackE(a,b)\\
          x + stackE(a,b) + unpE(r)\\
          x + stackE(a,b) + unpE(r)\\
          x + stackE(a,b) + unpE(r+r`)\\
          x + stackE(a,b)\\
          x + x'\\
          unpE(r)\\
          \hline
          \phantom{x}\\
          Minimum
        \end{tabular}
      \end{block}
    \end{column}
    \begin{column}{.31\textwidth}
      \begin{block}{\small Dot Bracket}\tabcolsep1mm
        \begin{tabular}{l}
          '('  +  '\ldots'  +  ')' \\
          '('  + x +  ')' \\
          '('  +  '\ldots'  + x +  ')' \\
          '('  + x +  '\ldots'  +  ')' \\
          '('  +  '\ldots'  + x +  '\ldots'  +  ')' \\
          '('  + x +  ')' \\
          x + x'\\
          '\ldots' \\
          \hline
          \phantom{x}\\
          Identity
        \end{tabular}
      \end{block}
    \end{column}
    \begin{column}{.16\textwidth}
      \begin{block}{\small Shape}\tabcolsep1mm
        \begin{tabular}{l}
          '$[$'  +  '$]$' \\
          x\\
          x\\
          x\\
          x\\
          '$[$'  + x +  '$]$' \\
          x + x'\\
          ''\\
          \hline
          \phantom{x}\\
          Identity
        \end{tabular}
      \end{block}
    \end{column}
  \end{columns}
  \small
  \begin{block}{RNAshapes -- Combination of these algebras}
    RNAshapes = (Energy, Dot Bracket, Shape)\\
    Choice funtion: Filter for identical shape and keep answer with lowest energy
  \end{block}
}


\section{RNAshapes for Alignments}
\subsection{}

\frame{
  \frametitle{\large Multiple Sequence Alignments as Input}
  \small
  \begin{block}{Problems \& Tasks}
    \begin{itemize}
      \item Handle multiple sequences and combine scores
      \item Check for basepairing (All, majority, at least 1)
      \item Non-standard base pairs (e.g. one sequence with A-C instead of G-C)
      \item Insertion in one sequence introduces long gap (energy depends on length)
      \item Covarying positions are of special interest
    \end{itemize}
  \end{block}
  \begin{block}{Solutions}
    \begin{itemize}
      \item Overall score = mean of individual scores
      \item Basepairing fraction defined by user (default: at least 1)
      \item Enhanced thermodynamic model:
        \begin{itemize}
        \item Non-standard base pairs (also those with gaps) get 0.0 kcal/mol
        \item Gap-aware handling of singlestranded regions (different from RNAalifold)
        \item Covariance score (reward covariance, penalize non-standard base pairs)
        \end{itemize}
    \end{itemize}
  \end{block}
}

\frame{
  \frametitle{Handling Alignments in ADP}
  \small
  \begin{block}{Grammar}
    \begin{itemize}
      \item The grammar does not work directly on the input, but on indexes.
      \item Grammar predicates (applied via ``with'') may work on the input.
    \end{itemize}
    $\Rightarrow$ A Grammar can be used for any kind of sequential data\\
    $\Rightarrow$ Predicates might need to be adapted for input.
  \end{block}
  \begin{block}{``basepairing'' predicate}
    For single sequence $S$:
    $$ basepairing(i,j) = \mbox{if }basepair (S[i],S[j]) = 1 \mbox{ then }true\mbox{ else }false\mbox{, where} $$
    $$ basepair (x,y) = \begin{cases}1, (x,y) \in \{(A,U),(U,A),(G,C),(C,G),(G,U),(U,G)\}\\0, \mbox{otherwise}\end{cases}$$
    For alignment $M$ (User-defined $CUT$-off):
    $$ basepairing(i,j) = if \frac{1}{N}\sum_{x \in M} basepair (x[i],x[j]) > CUT \mbox{ then }true\mbox{ else }false$$
  \end{block}
}

\frame{
  \frametitle{Algebras }
  \small
  \begin{block}{Algebra Adaptation}
    Algebras that do not directly utilise the input don't need adaptation.
    Only algebras for computing the free energy or partition function need to be changed.
  \end{block}
  \begin{block}{Energy}
    Energy $E$ of subword $i,j$ for alignment $M$ holding $N$ sequences:
    $$E_M(i,j) = \frac{1}{N} \sum_{x \in M} E_x(i,j)$$
    $\Rightarrow$ Individual energy function:\\
    For sequence $S$:
    $$SR(i,x,j) = x + sr\_energy(S[i],S[j])$$
    For alignment $M$:
    $$SR(i,x,j) = x + \frac{1}{N} \sum_{x \in M} sr\_energy(x[i],x[j])$$
  \end{block}
}

\frame{
  \frametitle{\large Gap-aware single-strand handling}
  \small
  \begin{block}{Example}
    \centering
    \texttt{CUUC\alert<2->{AUCAGUA.A}AAGCUUGGAGAAGAAUGAGCUUCAAUGAAAAGCUU\alert<2->{UGAAAGG}GAAC}\\
    \texttt{GCCU\alert<2->{AUGAC....}UACUUGUGCGGAGGGUGAUGCCGC.AGAUGUACAA\alert<2->{GGAAAGG}AGUC}\\
    \texttt{GCCC\alert<2->{AGGCAG...}AUGUUUUGUGGAGCCGCAACUCCAACACAGAACAU\alert<2->{UCAGGGG}GAGU}\\
    \texttt{AACU\alert<2->{AGGUAGU..}UCAAUCAGAGGAGCACAAACUCCAGCGAUGAUUGA\alert<2->{UGAGGGA}GAUU}\\
    \texttt{AAGC\alert<2->{AUGUAUUUG}GCGAGGUGUUAAGGAGAAGAACCUCCAAUACUCGC\alert<2->{UGAAGAA}GGUU}\\
    \texttt{((((\alert<2->{.........}((((((((((..(((.....)))..))))))))))\alert<2->{.......}))))}\\
  \end{block}
  \begin{block}<2->{Internal loop}
    \begin{itemize}
      \item Size of subword: 9 nt  $\Rightarrow$ asymmetric internal loop with 9 and 7 nt, resp.
      \item Actually: 5'-region: 5,6,7,8 or 9 nt, 3'-region: 7 nt
      \item This means: Mixture of symmetric and asymmetric internal loops with different loop lengths
    \end{itemize}
  \end{block}
  \begin{block}<2->{Handling gaps in unpaired regions}
    \begin{itemize}
      \item Different lengths: Recompute size of unpaired regions excluding gaps
      \item Gaps-only: Switch loop type (e.g. internal $\rightarrow$ bulge)
    \end{itemize}
  \end{block}
}

\frame{
  \frametitle{Covariance Scoring}
  \small
  \begin{block}{Covariation \& Inconsistency}
    \textbf{Covariation:} compensatory (A-U $\rightarrow$ G-C) and consistent (A-U $\rightarrow$ G-U)
    \begin{displaymath}
      C_{ij} = \sum_{a,b,a',b' \in \{A,C,G,U\}} f_{ij}(a,b) * D(a,b,a',b') * f_{ij}(a',b')
    \end{displaymath}
    \begin{displaymath}
      D(a,b,a',b') = \begin{cases}
        0, \mbox{not }(bp(a,b)|bp(a',b'))|(a,b)=(a',b')\\
        1, a=a'\ xor\ b=b'\\
        2, \mbox{otherwise}
      \end{cases}
    \end{displaymath}
    \textbf{Inconsistency:} Non-standard base pairs are allowed (0.0 kcal/mol) but need to be penalised. Gap-Gap pairs don't get penalised.
    \begin{displaymath}
      I_{ij} = \frac{1}{N} \sum_{x \in M} \begin{cases}0, x_i = x_j = gap|bp(x_i,x_j)\\1, \mbox{otherwise}\end{cases}
    \end{displaymath}
    \begin{center}$\Rightarrow$\textbf{\underline{Covariance Score:}} {\large $cv_{ij} = - C_{ij} + I_{ij}$}\end{center}
  \end{block}
}

\frame{
  \frametitle{\large RNAlishapes}
  \small
  \begin{block}{Implementation}
    \begin{itemize}
      \item Implemented in Haskell-ADP (Hopefully soon in C)
      \item Following analysis modes are supported:
        \begin{itemize}
          \item Optimal consensus structure
          \item Suboptimal consensus structures
          \item Shape analysis (5 abstraction levels)
          \item Boltzmann-weighted sampling (like SFOLD)
          \item Shape probabilities
        \end{itemize}
      \item User options:
        \begin{itemize}
          \item Weight of covariance score
          \item Minimum fraction of actual base pairs at pairing positions
        \end{itemize}
      \item Reads alignments in CLUSTALW format 
    \end{itemize}
  \end{block}
}

\section{Examples}
\subsection{}

\frame{
  \frametitle{tRNAs - Proof of Correctness}
  \begin{block}{Alignment and Predicted Consensus}
    \begin{figure}
      \subfigure[Rfam Alignment]{\includegraphics[width=\textwidth]{images/tRNA_example_ungap_ali_coloured.pdf}}
      \subfigure[RNAlishapes]{\includegraphics[width=.3\textwidth]{images/tRNA_example_structure_plot.pdf}}
      \hspace{2cm}
      \subfigure[RNAalifold]{\includegraphics[width=.3\textwidth]{images/tRNA_example_structure_plot.pdf}}
    \end{figure}
  \end{block}
}

\begin{frame}[fragile]
  \frametitle{tRNAs - Suboptimal Consensus Structures}
  \scriptsize
  \begin{block}{\small Energy range: 3 kcal/mol}
    \begin{verbatim}
       GCGUUCGUAGCUCAGUU-GGU--AGAGCAUCUGGUUUUGACCCUGAAUGUCAUGGGUUCGAAUCCCGUCGGUCGCG
-28.97 (((((((..((((...........))))..((((((...))))))......(((((.......)))))))))))).
-29.03 (((((((..((((...........))))..(((((.....)))))......(((((.......)))))))))))).
-30.31 (((((((..((((...........))))..((((.......))))......(((((.......)))))))))))).
-29.72 (((((((..((((.((...))...))))..((((.......))))......(((((.......)))))))))))).
-29.44 (((((((..(((.............)))..((((.......))))......(((((.......)))))))))))).
-30.15 (((((((..((((...........)))).(((((((...))))))).....(((((.......)))))))))))).
-30.21 (((((((..((((...........)))).((((((.....)))))).....(((((.......)))))))))))).
-31.49 (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).
-28.96 (((((((..((((...........)))).(((((.......))))).....((((.........))))))))))).
-29.56 (((((((..((((.((...))...)))).(((((((...))))))).....(((((.......)))))))))))).
-29.62 (((((((..((((.((...))...)))).((((((.....)))))).....(((((.......)))))))))))).
-30.9  (((((((..((((.((...))...)))).(((((.......))))).....(((((.......)))))))))))).
-29.28 (((((((..(((.............))).(((((((...))))))).....(((((.......)))))))))))).
-29.34 (((((((..(((.............))).((((((.....)))))).....(((((.......)))))))))))).
-30.62 (((((((..(((.............))).(((((.......))))).....(((((.......)))))))))))).
-29.43 (((((((..((((...........)))).(((((.......)))))......((((.......)))).))))))).
-28.84 (((((((..((((.((...))...)))).(((((.......)))))......((((.......)))).))))))).
-29.02 ((((((...((((...........))))..((((.......))))......(((((.......))))).)))))).
-28.86 ((((((...((((...........)))).(((((((...))))))).....(((((.......))))).)))))).
-28.92 ((((((...((((...........)))).((((((.....)))))).....(((((.......))))).)))))).
-30.2  ((((((...((((...........)))).(((((.......))))).....(((((.......))))).)))))).
-29.61 ((((((...((((.((...))...)))).(((((.......))))).....(((((.......))))).)))))).
-29.33 ((((((...(((.............))).(((((.......))))).....(((((.......))))).)))))).     
    \end{verbatim}
  \end{block}
\end{frame}

\begin{frame}[fragile]
  \frametitle{tRNAs - Suboptimal Shapes}
  \tiny
  \begin{block}{\small Most abstract shape, Energy range: 15 kcal/mol}
    \begin{verbatim}
        GCGUUCGUAGCUCAGUU-GGU--AGAGCAUCUGGUUUUGACCCUGAAUGUCAUGGGUUCGAAUCCCGUCGGUCGCG
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).  [[][][]]
-28.25  (((((((......................(((((.......))))).....(((((.......)))))))))))).  [[][]]
-25.89  (((((((((((((...........)))).(((((.......))))).))..(((((.......)))))))))))).  [[[][]][]]
-20.74  ((((((((((..((.((.((.((..(((.(((((.......)))))..)))..))..)))).)).)))))))))).  []
-19.11  .........((((...........)))).(((((.......))))).....(((((.......)))))........  [][][]
-18.5   ((((((...((((...........))))((((((.......)))).......((((.......)))))))))))).  [[][[][]]]
    \end{verbatim}
  \end{block}
  \begin{block}{\small Less abstract shape, Energy range: 7 kcal/mol}
    \begin{verbatim}
        GCGUUCGUAGCUCAGUU-GGU--AGAGCAUCUGGUUUUGACCCUGAAUGUCAUGGGUUCGAAUCCCGUCGGUCGCG
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-30.9   (((((((..((((.((...))...)))).(((((.......))))).....(((((.......)))))))))))).    [[[]][][]]
-28.25  (((((((......................(((((.......))))).....(((((.......)))))))))))).    [[][]]
-27.0   (((((((...................((.(((((.......)))))..)).(((((.......)))))))))))).    [[[]][]]
-26.1   ((((.((..((((...........)))).(((((.......))))).....(((((.......))))))).)))).    [[[][][]]]
-25.89  (((((((((((((...........)))).(((((.......))))).))..(((((.......)))))))))))).    [[[][]][]]
-25.86  (((((((..((((...........)))).((((.((...)).)))).....(((((.......)))))))))))).    [[][[]][]]
-25.51  ((((.((..((((.((...))...)))).(((((.......))))).....(((((.......))))))).)))).    [[[[]][][]]]
-25.33  (((((((..((((...........)))).(((((.......))))).....((.((.......)).))))))))).    [[][][[]]]
-25.3   (((((((((((((.((...))...)))).(((((.......))))).))..(((((.......)))))))))))).    [[[[]][]][]]
-25.27  (((((((..((((.((...))...)))).((((.((...)).)))).....(((((.......)))))))))))).    [[[]][[]][]]
-24.74  (((((((..((((.((...))...)))).(((((.......))))).....((.((.......)).))))))))).    [[[]][][[]]]
    \end{verbatim}
  \end{block}
\end{frame}

\begin{frame}[fragile]
  \frametitle{tRNAs - Boltzmann-weighted sampling}
  \tiny
  \begin{block}{\small 20 samples}
    \begin{verbatim}
        GCGUUCGUAGCUCAGUU-GGU--AGAGCAUCUGGUUUUGACCCUGAAUGUCAUGGGUUCGAAUCCCGUCGGUCGCG
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-28.49  .((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))..    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    [[][][]]
    \end{verbatim}
  \end{block}
\end{frame}

\begin{frame}[fragile]
  \frametitle{tRNAs - Shape probabilities}
  \tiny
  \begin{block}{\small Shape probabilities}
    \begin{verbatim}
        GCGUUCGUAGCUCAGUU-GGU--AGAGCAUCUGGUUUUGACCCUGAAUGUCAUGGGUUCGAAUCCCGUCGGUCGCG
-31.49  (((((((..((((...........)))).(((((.......))))).....(((((.......)))))))))))).    0.99606776      [[][][]]
-28.25  (((((((......................(((((.......))))).....(((((.......)))))))))))).    4.781712e-4     [[][]]
-25.89  (((((((((((((...........)))).(((((.......))))).))..(((((.......)))))))))))).    3.4460078e-3    [[[][]][]]
-22.38  (((((((..((...))...((.....)).(((((.......))))).....(((((.......)))))))))))).    7.973717e-6     [[][][][]]
    \end{verbatim}
  \end{block}
\end{frame}

\frame{
  \frametitle{Attenuators of bacterial \textit{trp}-operons}
  \small
  \begin{block}{Biology}
    \begin{itemize}
      \item Attenuation is important mechanism of gene regulation
      \item Formation of alternating structures
      \item Functions:
        \begin{itemize}
          \item Inhibition of translation initiation
          \item Premature termination of transcription
        \end{itemize}
    \end{itemize}
  \end{block}
  \begin{block}<only@1>{Example: \textit{trp}-operon}
    \begin{itemize}
      \item 8 leader regions
      \item Multiple sequence alignment, ClustalW
    \end{itemize}
  \end{block}
}

\frame{
  \frametitle{Attenuators of bacterial \textit{trp}-operons}
  \small
    \begin{block}{Shape analysis}
      \includegraphics[width=\textwidth]{images/trp_attenuator_ali_structure1_coloured.png}
    \end{block}
    \begin{block}{Results}
      \begin{itemize}
        \item Consensus A carries terminator hairpin (blue) and is energetically favourable
        \item Consensus B forms anti-terminator hairpin (red) and is less favourable
      \end{itemize}
    \end{block}
}

\frame{
  \frametitle{Low quality Alignment}
  \begin{block}{T-box sequences, Avg. PI 59\%}\centering
    \includegraphics[height=1.24\textheight]{images/t-box_alignment_structure_coloured.png}
  \end{block}
}

\section{Synopsis}
\subsection{}

\frame{
  \frametitle{Summary}
  \small
  \begin{block}{Algorithm}
    \begin{itemize}
      \item Structural analysis of aligned RNAs
      \item Combines shape abstraction and alignment folding
      \item Covariance Scoring
      \item Gap-aware thermodynamics
      \item User-defined pairing cut-off
    \end{itemize}
  \end{block}
  \begin{block}{Applications}
    \begin{itemize}
      \item Modes: MFE, suboptimal, shapes, sampling, shape porbabilities
      \item Structural features of RNA families, e.g.
        \begin{itemize}
          \item Robustness of MFE
          \item Switching
        \end{itemize}
      \item Improved predictions for low-quality, esp. gap-rich, alignments
    \end{itemize}
  \end{block}
}

\frame{
  \frametitle{Discussion}
  \small
  \begin{block}{Pros, Cons \& Outlook}
    \begin{itemize}
    \item Strong dependence on alignment quality\\$\Rightarrow$ Use MARNA?
    \item Computationally expensive
      \begin{itemize}
      \item Gap-counting
      \item Haskell implementation
      \end{itemize}
    \item Improve structure analysis and prediction
    \item Replace RNAalifold within RNAz
    \end{itemize}
  \end{block}
}

\frame
{
        \frametitle{\large Acknowledgements}
        \begin{itemize}
                \item Wolfgang Hess (Freiburg University)
                \item Robert Giegerich and Marc Rehmsmeier (Bielefeld University)
                \item Cyanolab people (Freiburg University)
        \end{itemize}
        \begin{center}
        \includegraphics[scale=0.35]{images/group2.jpg}
        \end{center}
}

\frame{
  \frametitle{}
  \begin{center}
    \Huge Thank You!\\
    \vspace{2cm}
    \movie[autostart,height=3.5cm,width=3.5cm]{}{images/att2_movie.avi}
  \end{center}
}
\end{document}